frontend/src/app/rating/rating.page.ts

```diff
@@ -0,0 +1,29 @@
+import { Component } from '@angular/core';
+import { Validators } from '@angular/forms';
+import { NavController } from '@ionic/angular';
+
+import { FormControl, FormGroup } from '../common/forms/forms';
+
+import { ApplicantScore, RatingForm } from './model/rating-form.model';
+
+@Component({
+  selector: 'app-rating',
+  templateUrl: './rating.page.html',
+  styleUrls: ['./rating.page.scss'],
+})
+export class RatingPage {
+  constructor(private navController: NavController) {}
+
+  ratingForm = new FormGroup<RatingForm>({
+    applicantScore: new FormGroup<ApplicantScore>({
+      rhetoric: new FormControl(undefined, Validators.required),
```

Comment:
I have a doubt. Why we need to define the type as undefined? Can't we use number? since the rhetoric type is defined as number in interface.

Reply:
Usually, if you don't want to pre-fill a form element with a value, then you simply leave it blank:
``` ts
rhetoric: new FormControl(),
```

But: Validators are passed as second parameter. So, if you want to use a Validator, you have to write:

``` ts
rhetoric: new FormControl(SOMETHING, Validators.required),
```

If you pass a number as first parameter, then the form element will be pre-filled with this value. And the Validator will automtically mark this Form Control as valid. This is pretty bad UX, since the user is now not forced to change the input value.

So, if you don't want to pre-fill the form element with a value, but still want to use a FormControl, then you need to do it like it was done here.